@nitmir python-ndg-httpsclient is not available on ubuntu 14.04 lts.
I updated python to 2.7.12 (as SNI was included in 2.7.9) and installed openssl, pyasn1 and ndg-httpsclient using pip .
Thing is, I checked SNI with a special check site, and it worked perfectly.
If I test with your code, it works as well, I get the same results as you.
But if I check my own server, which is behind a HAProxy, SNI is not recognized and I get a 503 error :

```
>>> import urllib2
>>> r = urllib2.urlopen('https://mysecreturl:443/cas/serviceValidate')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 503: Service Unavailable
```

However, using requests works perfectly.

--- a/cas.py
+++ b/cas.py
@@ -306,9 +306,9 @@ class CASClientWithSAMLV1(CASClientV2, SingleLogoutMixin):
             self.server_url, 'samlValidate',
         )
         return requests.post(
-            saml_validate_url, 
-            self.get_saml_assertion(ticket), 
-            params=params, 
+            saml_validate_url,
+            self.get_saml_assertion(ticket),
+            params=params,
             headers=headers)
 
     @classmethod

